### DISCUSSION TITLE
Dynamicly add command

### DISCUSSION THREAD

**Discussion by natir** [TwitchIO/TwitchIO#430] [Questions]
Hello thanks for your work.

I'd like to be able to load a certain command via a configuration file.

To do this, I need to create my functions while my script is running. One idea I've explored is to create anonymous functions and add them to my bot during launch, but it seems impossible to create anonymous asynchronous functions.

Do you have an idea for adding commands linked to a function not previously defined?

**Comment by IAmTomahawkx** [comment]
You could use [bot.command](https://twitchio.dev/en/stable/exts/commands.html#twitchio.ext.commands.Bot.command) to manually add commands. Typically this is a decorator, however it can also be used as a function. you don't really need to do anonymous functions,
```py
for command in command_list:
    async def newcommand(ctx):
        ...

    bot.command("name")(newcommand)
```
